### PR TITLE
refactor method definition in GlobalState

### PR DESCRIPTION
**Pull request by froydnj**
This change refactors the method definition in `GlobalState::initEmpty` to be somewhat less repetitive.  Whether this is better or not is a matter of taste, I suppose, but I think it's an improvement over mini blocks and `Loc::none()` everywhere.

### Motivation

I found `initEmpty` to be somewhat hard to follow; maybe this is an improvement.

### Test plan

See included automated tests.

### CODE REVIEW

File: core/GlobalState.cc [hunk 1]
```diff
@@ -113,6 +113,79 @@ unique_ptr<GlobalState> GlobalState::makeEmptyGlobalStateForHashing(spdlog::logg
...
+
+    // Auto-conversion when we're done.
+    operator MethodRef() {
+        return method;
+    }
+};
+
+MethodBuilder def(GlobalState &gs, ClassOrModuleRef klass, NameRef name) {
+    return MethodBuilder{gs, gs.enterMethodSymbol(Loc::none(), klass, name)};
+}
```
Comment (jez) [inline core/GlobalState.cc:187]: What do you think about giving this method a slightly longer / more descriptive name?

File: core/GlobalState.cc [hunk 2]
```diff
@@ -113,6 +113,79 @@ unique_ptr<GlobalState> GlobalState::makeEmptyGlobalStateForHashing(spdlog::logg
...
+        auto &arg = gs.enterMethodArgumentSymbol(Loc::none(), method, name);
+        arg.flags.isDefault = true;
+        return *this;
+    }
+
+    MethodBuilder &blockArg() {
+        auto &arg = gs.enterMethodArgumentSymbol(Loc::none(), method, Names::blkArg());
+        arg.flags.isBlock = true;
+        return *this;
+    }
```
Comment (jez) [inline core/GlobalState.cc:132]: Every method needs a block argument. What do you think about making the `.build()` be explicit, instead of happing on implicit conversion to a `MethodRef`, and moving this block arg stuff to that method?

**Review by jez — APPROVED** [review]
Generally love it. Two small suggestions.

File: core/GlobalState.cc (hunk 1 above)
Comment (froydnj) [reply core/GlobalState.cc]: WFM.  I was going for Ruby-ish, but I changed this to `enterMethod`.

File: core/GlobalState.cc (hunk 2 above)
Comment (froydnj) [reply core/GlobalState.cc]: Done.
